codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -203,6 +207,7 @@ func NewImpl(ctx {{.contextContext|raw}}, r Interface{{if .hasClass}}, classValu
 
 	lister := {{.type|lowercaseSingular}}Informer.Lister()
 
+	filterFunc := {{.controllerGetFilterFunc|raw}}(ctx)
```

Comment:
I'd rather see this come through the `controller.Options`, but the ordering and capture make this awkward.

I think that if you have `filterFunc := noFilter` up here, and optionally overwrite it below the capture should work out correctly though?

Reply:
in fact the rest of the things already seem to be doing this, so you are pretty close  🤔

Reply:
That's what @benmoss did in https://github.com/knative/pkg/pull/1909 too.

Reply:
#2170 rather 😄 . Just clicked your link and momentarily was very confused